rmgpy/rmg/reactors.py

```diff
@@ -0,0 +1,437 @@
+from rmgpy.thermo.thermodata import ThermoData
+from rmgpy.kinetics.arrhenius import Arrhenius, ArrheniusEP, ArrheniusBM, PDepArrhenius, MultiArrhenius, MultiPDepArrhenius
+from rmgpy.kinetics.kineticsdata import KineticsData
+from rmgpy.kinetics.falloff import Troe, ThirdBody, Lindemann
+from rmgpy.kinetics.chebyshev import Chebyshev
+from rmgpy.data.solvation import SolventData
+from rmgpy.kinetics.surface import StickingCoefficient
+from rmgpy.solver.termination import TerminationTime, TerminationConversion, TerminationRateRatio
+from rmgpy.data.kinetics.family import TemplateReaction
+from rmgpy.data.kinetics.depository import DepositoryReaction
+
+
+class PhaseSystem:
+    """
+    Class for tracking and managing all the phases and interfaces of species/reactions
+    needed to run a simulation
+    will typically have a core and edge PhaseSystem where the edge PhaseSystem
+    contains all species/reactions in the core PhaseSystem
+    """
+    def __init__(self, phases, interfaces):
+        self.phases = phases
+        self.interfaces = interfaces
+        self.species_dict = dict()
+        if 'nose' in sys.modules.keys(): #pyrms and nosetests aren't compatible
+            self.in_nose = True
+        else:
+            self.in_nose = False
+
+    def remove_species(self, spc):
+        """
+        If the species is found
+        removes that species and any reactions it is involved in
+        from the PhaseSystem and its Phases/Interfaces
+        """
+        for key, phase in self.phases.items():
+            rms_spc = phase.remove_species(spc.label)
+
+        if rms_spc:
+            for key, interface in self.interfaces.items():
+                interface.remove_species(rms_spc)
+
+            del self.species_dict[spc.label]
+
+    def add_reaction(self, rxn, species_list):
+        """
+        adds reaction to the appropriate phase/interface
+        within PhaseSystem
+        """
+        spclabels = [r.label for r in rxn.reactants+rxn.products]
+        phaseinv = []
+        for plabel, phase in self.phases.items():
+            out = [label for label in spclabels if label in phase.names]
+            if len(out) > 0:
+                phaseinv.append(phase)
+
+        if len(phaseinv) == 1:
+            phaseinv[0].add_reaction(rxn, species_list)
+        else:
+            phases = set(phaseinv)
+            for interface in self.interfaces:
+                if interface.phaseset == phases:
+                    interface.add_reaction(rxn, species_list)
+                    break
+
+    def pass_species(self, label, phasesys):
+        """
+        Adds a species from self to the input phase system phasesys
+        (usually adding an edge species to the core PhaseSystem)
+        also adds any reactions whose participating species are now all
+        in the core to the core PhaseSystem
+        reorders edge species ordering to match that of the core
+        """
+        phaselabel = ""
```

Comment:
Did this try to initialize for the `phase_label` below?

Reply:
Maybe in a prior iteration of the function XD. Removed.